requests-responses.md

```diff
@@ -113,46 +113,62 @@ The signature must be verified using the public key provided via `get_public_key
 
 ### Structure
 
-All requests have the following structure
-
+**All app requests have the following structure (like json-rpc 2.0)**
 ```tsx
 interface AppRequest {
 	method: string;
 	params: string[];
 	id: string;
 }
 ```
+Where 
+- method: name of the operation ('sendTransaction', 'singMessage', ...)
+- params: array of the operation specific parameters
+- id: identifier that allows to match requests and responses
 
-Wallet messages are
 
-```tsx
-type WalletEventName = 'disconnect';
 
-interface WalletEvent {
+**Wallet messages are operation responses or events.**
+```tsx
+interface WalletMessage {
 	type: WalletEventName | 'response';
-	<event-name>: <event-payload>
+	payload: <json-rpc response or event payload>;
 }
+
+type WalletEventName = 'disconnect' // currently only disconnect event is available
 ```
+Where 
+- type: name of the event ('disconnect', ...) or 'response'
+- payload: if type is name of some event then it is specific payload for the event, if type is 'response' it is json-rpc 2.0 response
 
-Exactly
 
+Exactly 
 ```tsx
-type WalletEvent = SendTransactionResponse | DisconnectEvent;
-```
+type WalletMessage = WalletResponse | WalletEvent;
 
-Where response is
+interface WalletEvent {
+    type: WalletEventName;
+    payload: <event-payload>; // specific payload for each event
+} 
+```
 
+Operation response is object with property `type` equals to `'response'` and property `payload` equals to json-rpc 2.0 response.
 ```tsx
-interface Response {
-	type: 'response';
-	response: {
-		result: unknown;
-		id: number;
-	} |
-	{
-		error: { code: number; message: string; data?: unknown };
-		id: number;
-	}
+interface WalletResponse {
+    type: 'response';
+    payload: OperationResult; // json-rpc 2.0 response
+}
+
+type OperationResult = OperationResultSuccess | OperationResultError;
```

Comment:
Why is this "operation"? we just had "response" above. Let's keep with it: ResponseResult{Success|Error}?

Also we don't do this at all. You simply have either `result` or `error` field without intermediate `payload` per json-rpc spec. See https://www.jsonrpc.org/specification